packages/react-core/src/components/FormSelect/__tests__/FormSelect.test.tsx

```diff
@@ -49,136 +51,147 @@ const groupedProps = {
-  expect(view).toMatchSnapshot();
-});
+  test('FormSelect input with no aria-label or id generates console error', () => {
+    const myMock = jest.fn() as any;
+    global.console = { error: myMock } as any;
 
+    render(
+      <FormSelect data-testid="test-id">
+        <FormSelectOption key={1} value={props.options[1].value} label={props.options[1].label} />
+      </FormSelect>
+    );
 
-test('required FormSelect input', () => {
-  const view = render(
-    <FormSelect required aria-label="required FormSelect">
-      <FormSelectOption key={1} value={props.options[1].value} label={props.options[1].label} />
-    </FormSelect>
-  );
-  expect(view.container).toMatchSnapshot();
-});
+    expect(screen.getByTestId('test-id').outerHTML).toMatchSnapshot();
+    expect(myMock).toBeCalled();
+  });
+
+  test('invalid FormSelect input', () => {
+    render(
+      <FormSelect validated="error" aria-label="invalid FormSelect">
+        <FormSelectOption key={1} value={props.options[1].value} label={props.options[1].label} />
+      </FormSelect>
+    );
+    expect(screen.getByLabelText('invalid FormSelect').outerHTML).toMatchSnapshot();
+  });
+
+  test('validated success FormSelect input', () => {
+    render(
+      <FormSelect validated={ValidatedOptions.success} aria-label="validated FormSelect">
+        <FormSelectOption key={1} value={props.options[1].value} label={props.options[1].label} />
+      </FormSelect>
+    );
+
+    const formSelect = screen.getByLabelText('validated FormSelect');
+
+    expect(formSelect.className).toContain('pf-m-success');
+    expect(formSelect.outerHTML).toMatchSnapshot();
+  });
+
+  test('validated error FormSelect input', () => {
+    render(
+      <FormSelect validated={ValidatedOptions.error} aria-label="validated FormSelect">
+        <FormSelectOption key={1} value={props.options[1].value} label={props.options[1].label} />
+      </FormSelect>
+    );
+    expect(screen.getByLabelText('validated FormSelect').outerHTML).toMatchSnapshot();
+  });
+
+  test('validated warning FormSelect input', () => {
+    render(
+      <FormSelect validated={ValidatedOptions.warning} aria-label="validated FormSelect">
+        <FormSelectOption key={1} value={props.options[1].value} label={props.options[1].label} />
+      </FormSelect>
+    );
+
+    const formSelect = screen.getByLabelText('validated FormSelect');
+
+    expect(formSelect.className).toContain('pf-m-warning');
+    expect(formSelect.outerHTML).toMatchSnapshot();
+  });
+
+  test('required FormSelect input', () => {
+    render(
+      <FormSelect required aria-label="required FormSelect">
+        <FormSelectOption key={1} value={props.options[1].value} label={props.options[1].label} />
+      </FormSelect>
+    );
+    expect(screen.getByLabelText('required FormSelect').outerHTML).toMatchSnapshot();
+  });
+
+  test('FormSelect passes value and event to onChange handler', () => {
+    const myMock = jest.fn();
```

Comment:
These are implementation details that this test shouldn't care about, but I can add at least that the mock was called with 'mr'. There is no longer a simulated event, so I can either grab the whole real event object;

```
{"_reactName": "onChange", "_targetInst": null, "bubbles": true, "cancelable": false, "currentTarget": null, "defaultPrevented": false, "eventPhase": 3, "isDefaultPrevented": [Function functionThatReturnsFalse], "isPropagationStopped": [Function functionThatReturnsFalse], "isTrusted": false, "nativeEvent": {"isTrusted": false}, "target": <select aria-invalid="false" aria-label="Some label" class="pf-c-form-control" data-ouia-component-id="OUIA-Generated-FormSelect-default-8" data-ouia-component-type="PF4/FormSelect" data-ouia-safe="true"><option class="" value="mr">Mr</option></select>, "timeStamp": 1648149977372, "type": "change"}
```

and paste that in, or exclude it since again, this is a detail we should not be concerned about. Rather when this test is rewritten it should verify a DOM change based on the selected option.